### PR TITLE
Fix db inconsistencies

### PR DESCRIPTION
**Pull request by jmcgettrick**
Hi everyone - I've closed all the related PRs and Issues and want to keep everything unified under this one request.

### CODE REVIEW

**Comment by jmcgettrick** [comment]
@yao9394 @golenkovm @aspark21 could you take a look at this and comment on anything that you spot please?

**Review by CharlotteSpinks — APPROVED** [review]

**Comment by golenkovm** [comment]
Hi,
Great. Thanks!
Cheers

**Comment by jmcgettrick** [comment]
Unfortunately I've had to remove the modifications to userid and courseid as those fields can't be updated without removing the indexes.

File: db/upgrade.php [hunk 1]
```diff
@@ -444,6 +444,62 @@ function xmldb_plagiarism_turnitin_upgrade($oldversion) {
...
+            $dbman->change_field_notnull($table, $field);
+        }
+
+        // Set ownerid to not allow null
+        $field = new xmldb_field('ownerid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, false, null, 'courseid');
+        if ($dbman->field_exists($table, $field)) {
+            $dbman->change_field_notnull($table, $field);
+        }
+
+        // Set turnitin_ctl to not allow null
```
Comment (dwinn) [inline db/upgrade.php:488]: If we set fields to not allow null, what happens if a database does already have null values for these fields? Is this something we need to consider? I think the update to the field might fail if null values are found.

I don't see why courseid, ownerid or turnitin_cid would ever be null but if there's a chance they could be then we need to consider it in this change.

Comment (jmcgettrick) [reply db/upgrade.php]: Yeah I see your point. They should never be null but they may, I've changed them accordingly!

**Review by dwinn — APPROVED** [review]

**Pull request merged**